src/utils/daos_hdlr.c

```diff
@@ -2343,239 +2342,289 @@ file_chmod(struct file_dfs *file_dfs, const char *path, mode_t mode)
 }
 
 static int
-fs_copy(struct file_dfs *src_file_dfs,
-	struct file_dfs *dst_file_dfs,
-	const char *dir_name,
-	int dfs_prefix_len,
-	const char *fs_dst_prefix)
+fs_copy_file(struct file_dfs *src_file_dfs,
+	     struct file_dfs *dst_file_dfs,
+	     struct stat *src_stat,
+	     const char *src_path,
+	     const char *dst_path)
 {
-	int	rc = 0;
-	DIR	*src_dir = NULL;
-	struct	stat st_dir_name;
-	char	*filename = NULL;
-	char	*dst_filename = NULL;
-	char	*next_path = NULL;
-	char	*next_dpath = NULL;
-
-
-	/* stat the source, and make sure it is a directory  */
-	rc = file_lstat(src_file_dfs, dir_name, &st_dir_name);
-	if (!S_ISDIR(st_dir_name.st_mode)) {
-		fprintf(stderr, "Source is not a directory: %s\n", dir_name);
+	int src_flags		= O_RDONLY;
+	int dst_flags		= O_CREAT | O_WRONLY;
+	mode_t tmp_mode_file	= S_IRUSR | S_IWUSR;
+	int rc;
+	uint64_t file_length	= src_stat->st_size;
+	uint64_t total_bytes	= 0;
+	uint64_t buf_size	= 64 * 1024 * 1024;
```

Comment:
would it make sense to allow this to be configurable? (not for this PR)

Reply:
Definitely. mpifileutils dcp/dsync allow this:
```
-b, --bufsize <SIZE>     - IO buffer size in bytes (default 4MB)
```